src/main/java/TaskList.java

```diff
@@ -0,0 +1,81 @@
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class TaskList {
+    ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> loadedData) {
+        this.taskList = loadedData;
+    }
+
+    public int size() {
+        return taskList.size();
+    }
+
+    public Task get(int index) {
+        return taskList.get(index);
+    }
+
+    public void add(Task task) {
+        taskList.add(task);
+    }
+
+    public void setDone(int index) {
+        Task newTask = this.taskList.get(index - 1).markAsDone();
+        this.taskList.set(index - 1, newTask);
+        Ui.printMarkAsDone(newTask);
+    }
+
+    public void setDelete(int index) {
```

Comment:
Do you think the naming for the methods can be improved on? e.g. `deleteTask`, `addTodo` etc because right now it follows a "set" prefix for completing, deleting, and adding Tasks which I find might be a little confusing 😅